internal/resourcehelper/helper.go

```diff
@@ -0,0 +1,251 @@
+package resourcehelper
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+//go:generate mockgen -source=helper.go -package=resourcehelper -destination=mock_helper_api.go
+
+type Helper interface {
+	IsNamespaced(kind string) bool
+	IsNotUpdateable(kind string) bool
+	NeedsResourceVersionUpdate(kind string) bool
+	UpdateResourceVersion(req *unstructured.Unstructured, found *unstructured.Unstructured) error
+	SetNodeSelectorTerms(obj *unstructured.Unstructured, terms map[string]string) error
+	IsOneTimer(obj *unstructured.Unstructured) (bool, error)
+	SetLabel(obj *unstructured.Unstructured, label string) error
+	SetMetaData(obj *unstructured.Unstructured, nm string, ns string)
+}
+
+func New() Helper {
+	return &resourceHelper{}
+}
+
+type resourceHelper struct{}
+
+func (rh resourceHelper) IsNamespaced(kind string) bool {
+	if kind == "Namespace" ||
+		kind == "ClusterRole" ||
+		kind == "ClusterRoleBinding" ||
+		kind == "SecurityContextConstraint" ||
+		kind == "SpecialResource" {
+		return false
+	}
+	return true
+}
+
+func (rh resourceHelper) IsNotUpdateable(kind string) bool {
+	// ServiceAccounts cannot be updated, maybe delete and create?
+	if kind == "ServiceAccount" || kind == "Pod" {
+		return true
+	}
+	return false
```

Comment:
I know you just wanted to extract functions, but how about we simplify those trivial methods while we're at it?
```suggestion
	return kind == "ServiceAccount" || kind == "Pod"
```